packages/web-components/src/wallet-connection/walletConnection.js

```diff
@@ -43,33 +43,53 @@ export const makeAgoricWalletConnection = async (chainStorageWatcher, rpc) => {
       }),
     );
 
+    let isFinished = false;
```

Comment:
a little hard to follow all the reads and mutations, but I can't think of anything clearer.